### ISSUE TITLE
Cypress run fails on Integration Tests with Jenkins CI erroring SIGILL

### ISSUE THREAD

**Issue by tejasviSri** [cypress-io/cypress#22570]
### Current behavior

Hi,

I'm trying to run Cypress in our CI pipeline and it fails on starting up the cypress

> [2022-06-28T17:10:57.879Z] + xvfb-run -a npm run cypress run

> [2022-06-28T17:10:58.135Z] 
> 
> [2022-06-28T17:10:58.136Z] > espresso-ui@1.0.0 cypress
> 
> [2022-06-28T17:10:58.136Z] > cypress open "run"
> 
> [2022-06-28T17:10:58.136Z] 
> 
> [2022-06-28T17:11:00.644Z] The Test Runner unexpectedly exited via a exit event with signal SIGILL
> 
> [2022-06-28T17:11:00.644Z] 
> 
> [2022-06-28T17:11:00.644Z] Please search Cypress documentation for possible solutions:
> 
> [2022-06-28T17:11:00.644Z] 
> 
> [2022-06-28T17:11:00.644Z] https://on.cypress.io/
> 
> [2022-06-28T17:11:00.644Z] 
> 
> [2022-06-28T17:11:00.644Z] Check if there is a GitHub issue describing this crash:
> 
> [2022-06-28T17:11:00.644Z] 
> 
> [2022-06-28T17:11:00.644Z] https://github.com/cypress-io/cypress/issues
> 
> [2022-06-28T17:11:00.644Z] 
> 
> [2022-06-28T17:11:00.644Z] Consider opening a new issue.
> 
> [2022-06-28T17:11:00.644Z] 
> 
> [2022-06-28T17:11:00.644Z] ----------
> 
> [2022-06-28T17:11:00.644Z] 
> 
> [2022-06-28T17:11:00.644Z] Platform: linux-x64 (Debian - 11.3)
> 
> [2022-06-28T17:11:00.644Z] Cypress Version: 10.0.2

### Cypress Version

10.0.2

### Package Manager

npm

### Operating system

Linux

### Other

We tried with 10.0.0 or 10.2.0 and the error was same

**Comment by emilyrohrbough** [comment]
@tejasviSri Can you provide more information to help us understand your setup and how to reproduce the issue? Providing the debug logs by use `DEBUG=cypress:*` would be helpful.

**Comment by Goutham-T** [comment]
hi @emilyrohrbough 

I am facing a similar issue. Please find the attached logs.

````
[2022-07-11T05:14:50.278Z] 2022-07-11T05:14:50.176Z cypress:cli:cli cli starts with arguments ["/usr/local/bin/node","/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482/node_modules/.bin/cypress","run","-b","chrome"]

[2022-07-11T05:14:50.278Z] 2022-07-11T05:14:50.177Z cypress:cli NODE_OPTIONS is not set

[2022-07-11T05:14:50.278Z] 2022-07-11T05:14:50.177Z cypress:cli:cli program parsing arguments

[2022-07-11T05:14:50.278Z] 2022-07-11T05:14:50.182Z cypress:cli:cli running Cypress with args [ Command { _events: [Object: null prototype] { 'option:browser': [Function (anonymous)], 'option:ci-build-id': [Function (anonymous)], 'option:component': [Function (anonymous)], 'option:config': [Function (anonymous)], 'option:config-file': [Function (anonymous)], 'option:e2e': [Function (anonymous)], 'option:env': [Function (anonymous)], 'option:group': [Function (anonymous)], 'option:key': [Function (anonymous)], 'option:headed': [Function (anonymous)], 'option:headless': [Function (anonymous)], 'option:no-exit': [Function (anonymous)], 'option:parallel': [Function (anonymous)], 'option:port': [Function (anonymous)], 'option:project': [Function (anonymous)], 'option:quiet': [Function (anonymous)], 'option:record': [Function (anonymous)], 'option:reporter': [Function (anonymous)], 'option:reporter-options': [Function (anonymous)], 'option:spec': [Function (anonymous)], 'option:tag': [Function (anonymous)], 'option:dev': [Function (anonymous)] }, _eventsCount: 22, _maxListeners: undefined, commands: [], options: [ [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option] ], parent: Command { _events: [Object: null prototype], _eventsCount: 1, _maxListeners: undefined, commands: [Array], options: [Array], parent: null, _allowUnknownOption: false, _args: [], rawArgs: [Array], _scriptPath: '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482/node_modules/.bin/cypress', _name: 'cypress', _optionValues: {}, _storeOptionsAsProperties: true, _passCommandToAction: true, _actionResults: [], _actionHandler: null, _executableHandler: false, _executableFile: null, _defaultCommandName: null, _exitCallback: null, _aliases: [], _hidden: false, _helpFlags: '-h, --help', _helpDescription: 'display help for command', _helpShortFlag: '-h', _helpLongFlag: '--help', _hasImplicitHelpCommand: undefined, _helpCommandName: 'help', _helpCommandnameAndArgs: 'help [command]', _helpCommandDescription: 'display help for command', _usage: '<command> [options]', args: [Array], [Symbol(kCapture)]: false }, _allowUnknownOption: false, _args: [], rawArgs: null, _scriptPath: null, _name: 'run', _optionValues: {}, _storeOptionsAsProperties: true, _passCommandToAction: true, _actionResults: [], _actionHandler: [Function: listener], _executableHandler: false, _executableFile: null, _defaultCommandName: null, _exitCallback: null, _aliases: [], _hidden: false, _helpFlags: '-h, --help', _helpDescription: 'display help for command', _helpShortFlag: '-h', _helpLongFlag: '--help', _hasImplicitHelpCommand: 0, _helpCommandName: 'help', _helpCommandnameAndArgs: 'help [command]', _helpCommandDescription: 'display help for command', _usage: '[options]', _description: 'Runs Cypress tests from the CLI without the GUI', _argsDescription: undefined, exit: true, browser: 'chrome', args: [], [Symbol(kCapture)]: false } ]

[2022-07-11T05:14:50.538Z] 2022-07-11T05:14:50.316Z cypress:cli:cli variable-length opts parsed { args: [ '/usr/local/bin/node', '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482/node_modules/.bin/cypress', 'run', '-b', 'chrome' ], opts: Command { _events: [Object: null prototype] { 'option:browser': [Function (anonymous)], 'option:ci-build-id': [Function (anonymous)], 'option:component': [Function (anonymous)], 'option:config': [Function (anonymous)], 'option:config-file': [Function (anonymous)], 'option:e2e': [Function (anonymous)], 'option:env': [Function (anonymous)], 'option:group': [Function (anonymous)], 'option:key': [Function (anonymous)], 'option:headed': [Function (anonymous)], 'option:headless': [Function (anonymous)], 'option:no-exit': [Function (anonymous)], 'option:parallel': [Function (anonymous)], 'option:port': [Function (anonymous)], 'option:project': [Function (anonymous)], 'option:quiet': [Function (anonymous)], 'option:record': [Function (anonymous)], 'option:reporter': [Function (anonymous)], 'option:reporter-options': [Function (anonymous)], 'option:spec': [Function (anonymous)], 'option:tag': [Function (anonymous)], 'option:dev': [Function (anonymous)] }, _eventsCount: 22, _maxListeners: undefined, commands: [], options: [ [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option], [Option] ], parent: Command { _events: [Object: null prototype], _eventsCount: 1, _maxListeners: undefined, commands: [Array], options: [Array], parent: null, _allowUnknownOption: false, _args: [], rawArgs: [Array], _scriptPath: '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482/node_modules/.bin/cypress', _name: 'cypress', _optionValues: {}, _storeOptionsAsProperties: true, _passCommandToAction: true, _actionResults: [], _actionHandler: null, _executableHandler: false, _executableFile: null, _defaultCommandName: null, _exitCallback: null, _aliases: [], _hidden: false, _helpFlags: '-h, --help', _helpDescription: 'display help for command', _helpShortFlag: '-h', _helpLongFlag: '--help', _hasImplicitHelpCommand: undefined, _helpCommandName: 'help', _helpCommandnameAndArgs: 'help [command]', _helpCommandDescription: 'display help for command', _usage: '<command> [options]', args: [Array], [Symbol(kCapture)]: false }, _allowUnknownOption: false, _args: [], rawArgs: null, _scriptPath: null, _name: 'run', _optionValues: {}, _storeOptionsAsProperties: true, _passCommandToAction: true, _actionResults: [], _actionHandler: [Function: listener], _executableHandler: false, _executableFile: null, _defaultCommandName: null, _exitCallback: null, _aliases: [], _hidden: false, _helpFlags: '-h, --help', _helpDescription: 'display help for command', _helpShortFlag: '-h', _helpLongFlag: '--help', _hasImplicitHelpCommand: 0, _helpCommandName: 'help', _helpCommandnameAndArgs: 'help [command]', _helpCommandDescription: 'display help for command', _usage: '[options]', _description: 'Runs Cypress tests from the CLI without the GUI', _argsDescription: undefined, exit: true, browser: 'chrome', args: [], [Symbol(kCapture)]: false } }

[2022-07-11T05:14:50.538Z] 2022-07-11T05:14:50.317Z cypress:cli parsed cli options { browser: 'chrome' }

[2022-07-11T05:14:50.538Z] 2022-07-11T05:14:50.318Z cypress:cli verifying Cypress app

[2022-07-11T05:14:50.538Z] 2022-07-11T05:14:50.318Z cypress:cli Using CYPRESS_CACHE_FOLDER from environment variable

[2022-07-11T05:14:50.538Z] 2022-07-11T05:14:50.318Z cypress:cli Using CYPRESS_CACHE_FOLDER from environment variable

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.318Z cypress:cli using environment variable CYPRESS_CACHE_FOLDER /root/.cache/Cypress

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.319Z cypress:cli checking environment variables

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.328Z cypress:cli checking if executable exists /root/.cache/Cypress/10.3.0/Cypress/Cypress

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.331Z cypress:cli Binary is executable? : true

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.332Z cypress:cli binaryDir is  /root/.cache/Cypress/10.3.0/Cypress

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.332Z cypress:cli Reading binary package.json from: /root/.cache/Cypress/10.3.0/Cypress/resources/app/package.json

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.338Z cypress:cli Found binary version 10.3.0 installed in: /root/.cache/Cypress/10.3.0/Cypress

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.339Z cypress:cli { verified: true }

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.339Z cypress:cli is Verified ? true

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.339Z cypress:cli:run processing run options { browser: 'chrome', key: null, spec: null, reporter: null, reporterOptions: null, project: '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482' }

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.339Z cypress:cli:run --key is not set, looking up environment variable CYPRESS_RECORD_KEY

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.340Z cypress:cli:run run to spawn.start args ["--run-project","/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482","--browser","chrome"]

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.340Z cypress:cli undefined DISPLAY environment variable

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.340Z cypress:cli Cypress will spawn its own Xvfb

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.340Z cypress:cli Using CYPRESS_CACHE_FOLDER from environment variable

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.340Z cypress:cli Using CYPRESS_CACHE_FOLDER from environment variable

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.340Z cypress:cli using environment variable CYPRESS_CACHE_FOLDER /root/.cache/Cypress

[2022-07-11T05:14:50.539Z] 2022-07-11T05:14:50.341Z cypress:cli needs to start own Xvfb? true

[2022-07-11T05:14:50.540Z] 2022-07-11T05:14:50.341Z cypress:cli Starting Xvfb

[2022-07-11T05:14:50.540Z] 2022-07-11T05:14:50.447Z cypress:cli spawning, should retry on display problem? false

[2022-07-11T05:14:50.540Z] 2022-07-11T05:14:50.455Z cypress:cli passing DISPLAY :99

[2022-07-11T05:14:50.540Z] 2022-07-11T05:14:50.455Z cypress:cli spawn args [ '--no-sandbox', '--', '--run-project', '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', '--browser', 'chrome', '--cwd', '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', '--userNodePath', '/usr/local/bin/node', '--userNodeVersion', '16.14.2' ] { detached: false, stdio: [ 'inherit', 'inherit', 'pipe' ] }

[2022-07-11T05:14:50.540Z] 2022-07-11T05:14:50.455Z cypress:cli spawning Cypress with executable: /root/.cache/Cypress/10.3.0/Cypress/Cypress

[2022-07-11T05:14:50.540Z] 2022-07-11T05:14:50.460Z cypress:cli piping child STDERR to process STDERR

[2022-07-11T05:14:51.114Z]   cypress:ts registering ts-node on directory undefined +0ms

[2022-07-11T05:14:51.381Z]   cypress:ts Running without ts-node hook in environment "production" +103ms

[2022-07-11T05:14:53.293Z]   cypress:server:appdata path: /root/.config/Cypress/cy/production/cache +0ms

[2022-07-11T05:14:53.294Z]   cypress:server:appdata path: /root/.config/Cypress/cy/production/browsers +97ms

[2022-07-11T05:14:53.294Z]   cypress:server:cypress starting cypress with argv [ '/root/.cache/Cypress/10.3.0/Cypress/Cypress', '--no-sandbox', '--', '--run-project', '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', '--browser', 'chrome', '--cwd', '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', '--userNodePath', '/usr/local/bin/node', '--userNodeVersion', '16.14.2' ] +0ms

[2022-07-11T05:14:53.294Z]   cypress:server:args argv array: [ '/root/.cache/Cypress/10.3.0/Cypress/Cypress', '--no-sandbox', '--run-project', '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', '--browser', 'chrome', '--cwd', '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', '--userNodePath', '/usr/local/bin/node', '--userNodeVersion', '16.14.2' ] +0ms

[2022-07-11T05:14:53.294Z]   cypress:server:args parsed argv options { options: { _: [ '/root/.cache/Cypress/10.3.0/Cypress/Cypress' ], sandbox: false, 'run-project': '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', runProject: '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', browser: 'chrome', cwd: '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', userNodePath: '/usr/local/bin/node', userNodeVersion: '16.14.2' } } +2ms

[2022-07-11T05:14:53.294Z]   cypress:server:args argv parsed: { _: [ '/root/.cache/Cypress/10.3.0/Cypress/Cypress' ], sandbox: false, runProject: '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', browser: 'chrome', cwd: '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', userNodePath: '/usr/local/bin/node', userNodeVersion: '16.14.2', invokedFromCli: true } +3ms

[2022-07-11T05:14:53.294Z]   cypress:server:util:proxy found proxy environment variables { npm_config_noproxy: '' } +0ms

[2022-07-11T05:14:53.294Z]   cypress:server:args options { _: [ '/root/.cache/Cypress/10.3.0/Cypress/Cypress' ], sandbox: false, runProject: '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', browser: 'chrome', cwd: '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', userNodePath: '/usr/local/bin/node', userNodeVersion: '16.14.2', invokedFromCli: true, config: {} } +2ms

[2022-07-11T05:14:53.295Z]   cypress:server:args argv options: { _: [ '/root/.cache/Cypress/10.3.0/Cypress/Cypress' ], sandbox: false, runProject: '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', browser: 'chrome', cwd: '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', userNodePath: '/usr/local/bin/node', userNodeVersion: '16.14.2', invokedFromCli: true, config: {}, projectRoot: '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482' } +0ms

[2022-07-11T05:14:53.295Z]   cypress:server:cypress from argv [ '/root/.cache/Cypress/10.3.0/Cypress/Cypress', '--no-sandbox', '--run-project', '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', '--browser', 'chrome', '--cwd', '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', '--userNodePath', '/usr/local/bin/node', '--userNodeVersion', '16.14.2' ] got options { _: [ '/root/.cache/Cypress/10.3.0/Cypress/Cypress' ], sandbox: false, runProject: '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', browser: 'chrome', cwd: '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482', userNodePath: '/usr/local/bin/node', userNodeVersion: '16.14.2', invokedFromCli: true, config: {}, projectRoot: '/home/jenkins/agent/workspace/main_Test_test-ui-cypress_PR-482' } +10ms

[2022-07-11T05:14:53.295Z]   cypress:server:cypress scaling electron app in headless mode +1ms

[2022-07-11T05:14:53.295Z] 2022-07-11T05:14:53.154Z cypress:cli child event fired { event: 'exit', code: null, signal: 'SIGILL' }

[2022-07-11T05:14:53.295Z] 2022-07-11T05:14:53.155Z cypress:cli detecting arch { osPlatform: 'linux', osArch: 'x64' }

[2022-07-11T05:14:53.295Z] 2022-07-11T05:14:53.172Z cypress:cli arm uname -m result: { stdout: 'x86_64' } 

[2022-07-11T05:14:53.295Z] 2022-07-11T05:14:53.173Z cypress:cli child event fired { event: 'close', code: null, signal: 'SIGILL' }

[2022-07-11T05:14:53.295Z] 2022-07-11T05:14:53.188Z cypress:cli Stopping Xvfb

[2022-07-11T05:14:53.295Z] The Test Runner unexpectedly exited via a exit event with signal SIGILL

[2022-07-11T05:14:53.295Z] 

[2022-07-11T05:14:53.295Z] Please search Cypress documentation for possible solutions:

[2022-07-11T05:14:53.295Z] 

[2022-07-11T05:14:53.295Z] https://on.cypress.io/

[2022-07-11T05:14:53.295Z] 

[2022-07-11T05:14:53.295Z] Check if there is a GitHub issue describing this crash:

[2022-07-11T05:14:53.295Z] 

[2022-07-11T05:14:53.296Z] https://github.com/cypress-io/cypress/issues

[2022-07-11T05:14:53.296Z] 

[2022-07-11T05:14:53.296Z] Consider opening a new issue.

[2022-07-11T05:14:53.296Z] 

[2022-07-11T05:14:53.296Z] ----------

[2022-07-11T05:14:53.296Z] 

[2022-07-11T05:14:53.296Z] Platform: linux-x64 (Debian - 11.3)

[2022-07-11T05:14:53.296Z] Cypress Version: 10.3.0

[2022-07-11T05:14:53.867Z] 

[2022-07-11T05:14:53.867Z] > test-ui-cypress@1.0.0 clean:env

[2022-07-11T05:14:53.867Z] > unset CYPRESS_ENVIRONMENT && unset CYPRESS_BASE_URL

````

let me know if you need more info to fix this issue.

**Comment by emilyrohrbough** [comment]
@Goutham-T Please provide more information on the command you are running, OS, Cypress version, node version, etc. The debugs logs wasn't enough for me to reproduce the issue.

**Comment by Goutham-T** [comment]
@emilyrohrbough 

please find the information below:

````
Command: cypress run -b chrome --headed  (in a shell command)

OS - LINUX

Platform: linux-x64 (Debian - 11.3)

Node version - 16.14.2

Cypress version - 10.3.0

````

**Comment by mjhenkes** [comment]
@achs0 This issue is remarkably similar to https://github.com/cypress-io/cypress/issues/6713, have you tried to unset the DISPLAY env as suggested [here](https://github.com/cypress-io/cypress/issues/6713#issuecomment-611958774)?

**Comment by achs0** [comment]
> @achs0 This issue is remarkably similar to #6713, have you tried to unset the DISPLAY env as suggested [here](https://github.com/cypress-io/cypress/issues/6713#issuecomment-611958774)?

This is not my issue 😅

**Comment by AtofStryker** [comment]
@Goutham-T @tejasviSri any idea if you are able to reproduce this outside of Jenkins, maybe in a Docker image? I am wondering if the application has permissions for the child process or if it is running out of memory, or maybe neither 😅 . Have you tried running [gdb](https://www.tutorialspoint.com/gnu_debugger/installing_gdb.htm) to get a better idea of what is happening?

**Comment by coryasilva** [comment]
@AtofStryker thanks for the reply.  Finally had time to come back to this; fixed it with permission escalation as you suggested.

Here is my kubernetes pod yaml for the Jenkins stage.

```yaml
apiVersion: v1
kind: Pod
spec:
  containers:
    - name: cypress
      image: cypress/included:10.3.0
      command:
        - "sleep"
      args:
        - "9999999"
      securityContext:
        privileged: true
        runAsGroup: 1000
        runAsUser: 1000
      resources:
        limits:
          cpu: "1000m"
          memory: "3Gi"
        requests:
          cpu: "1000m"
          memory: "2Gi"
```

**Comment by AtofStryker** [comment]
@coryasilva that's awesome! I am glad it fixed your issue. I am going to close this but let us know if you run into other troubles.
